### PR TITLE
Missing std_msgs depend in simple_bag_recorder sample

### PR DESCRIPTION
**Pull request by j-rivero**
https://github.com/ros2/rosbag2/pull/800 added the simple_bag_recorder file, which [uses `std_msgs`](https://github.com/ros2/rosbag2/pull/800/files#diff-944bba4d88aecb5c0e6da7da34a658cb39b5d9cd8a9124f144b13db55cd55f4dR17) but does not declare it in package.xml making CI to cry: [Fci__nightly-release_ubuntu_focal_amd64](https://build.ros2.org/view/Fci/job/Fci__nightly-release_ubuntu_focal_amd64/309/) and [Fci__nightly-fastrtps_ubuntu_focal_amd64](https://build.ros2.org/view/Fci/job/Fci__nightly-fastrtps_ubuntu_focal_amd64/318/).

I'm also alpha-order the depends in the PR.

### CODE REVIEW

**Review by emersonknapp — APPROVED** [review]
Thanks for the patch!

**Comment by asobhy-qnx** [comment]
This patch is incomplete. There are changes that must be added in the CMakeLists.txt as well.

You will need to add 

find_package(std_msgs REQUIRED)

and you will need to change 

ament_target_dependencies(simple_bag_recorder
  rclcpp
  rosbag2_cpp)

ament_target_dependencies(simple_bag_recorder
  std_msgs
  rclcpp
  rosbag2_cpp)

**Comment by j-rivero** [comment]
> This patch is incomplete. There are changes that must be added in the CMakeLists.txt as well.

Thanks, d2a005b

**Review by emersonknapp — APPROVED** [review]

**Review by Blast545 — APPROVED** [review]
LGTM @j-rivero, thanks! Should we trigger a custom windows CI build to check it solves the issue before merging?

**Comment by emersonknapp** [comment]
Yes - do not merge without Jenkins CI

**Comment by Blast545** [comment]
CI configured for release up to `bag_recorder_nodes`:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/15169/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/9896/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/12856/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/15350/)

**Comment by nuclearsandwich** [comment]
The above CI run was slightly misconfigured, the CI_SCRIPTS_BRANCH parameter was left blank rather than its default of `master`. I occasionally get the CI_BRANCH_TO_TEST and CI_SCRIPTS_BRANCH parameters backwards and maybe that is what happened but the default was not restored when it was caught and corrected.

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/15172/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/9899/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/12859/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/15353/)

**Comment by nuclearsandwich** [comment]
>    * Windows [Build Status](http://ci.ros2.org/job/ci_windows/15358/)

The CMake warnings are present in the latest Foxy patch release CI https://ci.ros2.org/job/ci_windows/15323/ (stable distributions do not receive linter- or warning-only backports. I have triggered https://ci.ros2.org/job/ci_windows/15360/ to verify that the MSBuild warnings are present in mainline foxy as well.

**Comment by nuclearsandwich** [comment]
> I have triggered [ci.ros2.org/job/ci_windows/15360](https://ci.ros2.org/job/ci_windows/15360/) to verify that the MSBuild warnings are present in mainline foxy as well.

Well the msbuild warnings weren't in the mainline build but it could still be visual studio differences if there was a recent update.

**Comment by emersonknapp** [comment]
Yes the warnings are not even in this package - they are all in dependencies below.

**Pull request merged**
